Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -15,38 +21,134 @@
 /**
  * Represents an anchor of a Block that can connect to (1 or more) Connections.
  * 
+ * A ConnectionAnchor has an invisible part that acts as an enlargement of the touch zone.
+ * 
  * The primary Connection (if present) is the first element in getConnections().
  * This means that the oldest Connection is the primary connection.
  */
-public abstract class ConnectionAnchor extends Circle implements ComponentLoader {
-    /** The pane on which this Anchor resides. */
-    private CustomUIPane pane;
+public abstract class ConnectionAnchor extends StackPane implements ComponentLoader {
 
-    /** The block this Anchor is connected to. */
+    /** The block this ConnectionAnchor belongs to. */
     private Block block;
 
     /** The connections this anchor has, can be empty for no connections. */
     private List<Connection> connections;
+    
+    /** The visual representation of the ConnectionAnchor. */
+    @FXML private Shape visibleAnchor;
+    
+    /** The invisible part of the ConnectionAnchor (the touch zone). */
+    @FXML private Shape invisibleAnchor;
+    
+    /** Property storing the error state. */
+    private BooleanProperty errorState;
+    
+    /**
+     * Property storing the active state.
+     * When active, the ConnectionAnchor will react to user input.
```

Comment:
So, this is like `disabled` in JavaFX, not like `active` in the TactilePane sense?